### DISCUSSION TITLE
Build-> Core -> KnowledgeMW (Build failed)

### DISCUSSION THREAD

**Discussion by shounakacharya** [Sunbird-Ed/Community#79] [Installation] [answered]
System Configuration Details:
1.	KP, DP, DB, Yarn, Jenkins, Other-Services: Ubuntu 18.04 LTS, 4vCPU, 16GB RAM, 250 GB HDD
2.	Google Kubernetes engine: 1.21.12-gke.2200 with 4 worker nodes. Note: Currently Sunbird is not compatible with K8s APIs  >= 1.22 and we have checked this
3.	All systems are running on GCP
4. Github Release Tag: release-4.3.0_RC1

@gandham-santhosh and  @keshavprasadms

Jenkins Logs:

Started by user [Admin Deployer](http://35.222.148.131:8080/user/admin)
Rebuilds build #[1](http://35.222.148.131:8080/job/Build/job/Core/job/KnowledgeMW/1)
Checking out git https://github.com/project-sunbird/knowledge-mw-service.git into /var/lib/jenkins/workspace/Build/Core/KnowledgeMW@script to read Jenkinsfile
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Build/Core/KnowledgeMW@script/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/knowledge-mw-service.git # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/knowledge-mw-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/knowledge-mw-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.3.0_RC1^{commit} # timeout=10
 > git rev-parse release-4.3.0_RC1^{commit} # timeout=10
Checking out Revision c8c92fd9f48b9f85c3262c1239205db09abd24d4 (release-4.3.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f c8c92fd9f48b9f85c3262c1239205db09abd24d4 # timeout=10
Commit message: "Merge pull request #267 from keshavprasadms/release-4.3.0"
 > git rev-list --no-walk c8c92fd9f48b9f85c3262c1239205db09abd24d4 # timeout=10
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] Start of Pipeline
[Pipeline] node
Running on [Jenkins](http://35.222.148.131:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/Build/Core/KnowledgeMW
[Pipeline] {
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (Checkout)
[Pipeline] echo
Found environment variable named hub_org with value as: shounakacharya
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/project-sunbird/knowledge-mw-service.git
 > git init /var/lib/jenkins/workspace/Build/Core/KnowledgeMW # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/knowledge-mw-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/knowledge-mw-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/knowledge-mw-service.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-4.3.0_RC1^{commit} # timeout=10
 > git rev-parse release-4.3.0_RC1^{commit} # timeout=10
Checking out Revision c8c92fd9f48b9f85c3262c1239205db09abd24d4 (release-4.3.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f c8c92fd9f48b9f85c3262c1239205db09abd24d4 # timeout=10
Commit message: "Merge pull request #267 from keshavprasadms/release-4.3.0"
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ echo release-4.3.0_RC1_c8c92fd_2
[Pipeline] echo
build_tag: release-4.3.0_RC1_c8c92fd_2
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build)
[Pipeline] echo
Environment will be : build
[Pipeline] sh
+ git submodule update --init
Submodule 'src/libs' (https://github.com/project-sunbird/sunbird-js-utils) registered for path 'src/libs'
Cloning into '/var/lib/jenkins/workspace/Build/Core/KnowledgeMW/src/libs'...
Submodule path 'src/libs': checked out 'f9580a929511b7c735dca3e3b7cd17f955616d19'
[Pipeline] sh
+ chmod 777 build.sh
[Pipeline] sh
+ ./build.sh release-4.3.0_RC1_c8c92fd_2 master shounakacharya
--2022-08-19 12:34:20--  https://www.imagemagick.org/download/binaries/ImageMagick-i386-pc-solaris2.11.tar.gz
Resolving www.imagemagick.org (www.imagemagick.org)... 50.251.58.13
Connecting to www.imagemagick.org (www.imagemagick.org)|50.251.58.13|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://download.imagemagick.org/archive/binaries/ImageMagick-i386-pc-solaris2.11.tar.gz [following]
--2022-08-19 12:34:20--  https://download.imagemagick.org/archive/binaries/ImageMagick-i386-pc-solaris2.11.tar.gz
Resolving download.imagemagick.org (download.imagemagick.org)... 50.251.58.13
Connecting to download.imagemagick.org (download.imagemagick.org)|50.251.58.13|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-08-19 12:34:21 ERROR 404: Not Found.

[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 8
Finished: FAILURE

**Comment by beepdot** [comment, marked as answer]
Oops. It looks like the image magic binary url is no more valid as the image magic provider might have deprecated the url. Please make changes to the [build script](https://github.com/project-sunbird/knowledge-mw-service/blob/release-4.3.0/build.sh#L9) and change the URL to -
https://imagemagick.org/archive/binaries/ImageMagick-i386-pc-solaris2.11.tar.gz

**Reply by shounakacharya** [reply]
@keshavprasadms ... where would this location be? On the local jenkins? and rerun the command again manually?

**Reply by beepdot** [reply]
You ca fork the repo, edit the file and then build the job from your forked repo.
Or you can also edit in jenkins workspace and execute from terminal the same set of commands that the job would run.

**Reply by shounakacharya** [reply]
@keshavprasadms : This worked. Thanks. I ran it from the Jenkins Workspace directly. Is there any ArtifactUpload Job that runs post this? If yes, what should be the parameters in that job to run it?

**Reply by beepdot** [reply]
Yes there is an artifact upload job which will push the docker image to your repo. That job looks for file named `metadata.json` from the Build job. If you have this file in your jenkins workspace, then you can go to the last failed job of the Build -> Click Replay and comment out all these lines https://github.com/project-sunbird/knowledge-mw-service/blob/release-4.3.0/Jenkinsfile#L10-L31 and click Run
Now the build job will archive the `metadata.json` file and your ArtifactUpload job should trigger itself.
This is a workaround to just make the job succeed and archive the json file.

**Reply by shounakacharya** [reply]
@keshavprasadms: thanks. I again ran a build. Let it fail, updated the build.sh and in the replay commented the checkout stage and it worked. Thanks. Marking this as answered.
